backend/dal/Helpers/Extensions/DbContextExtensions.cs

```diff
@@ -49,7 +50,8 @@ public static void UpdateProjectNumber(this PimsContext context, Project project
                 else if (p.PropertyType == PropertyTypes.Building && p.Building == null)
                     p.Building = context.Buildings.Find(p.BuildingId);
 
-                context.Update(p.UpdateProjectNumber(projectNumber));
+                context.Update(p.UpdateProjectNumbers(projectNumber));
```

Comment:
this logic is what gets called when a project is approved, so we need to remove the previous project number in addition to adding the new one.

Reply:
the old draft project number is the originalProjectNumber below, the new one is the one above.

Reply:
But you're telling it to update the same object twice.  The first time it will have the extra project number.  The second time you remove it.  Essentially the first time is irrelevant.

Reply:
I get the originalProjectNumber before it is set to the new project number, and then use that to do the removal, so originalProjectNumber != projectNumber

Reply:
Just to put this to bed here are the results of a debug:
The initial state of projectNumbers in the db is: ["DRAFT-00043"]
1) on line 42. originalProjectNumber is set to project.ProjectNumber which is "DRAFT-00043"
2) on line 43 project.ProjectNumber is updated to the new projectNumber "SPP-10032"
3) on line 53 updateProjectNumbers is called, with projectNumber which has the value of "SPP-10032"
4) on line 54 removeProjectNumber is called, with originalProjectNumber which has the value of "DRAFT-00043"
this results in the following projectNumbers field in the db: ["SPP-10032"]